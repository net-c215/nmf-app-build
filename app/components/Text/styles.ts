import { StyleSheet } from "react-native";
import colors from "../../style/colors";
import fonts from "../../style/fonts";

export default StyleSheet.create({
  darkGray: {
    color: colors.darkLink70
  },
  lightGray: {
    color: colors.swordGray
  },
  bold: {
    fontFamily: fonts.Bold
  }
});
