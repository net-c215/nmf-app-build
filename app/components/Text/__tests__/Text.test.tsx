import React from "react";
import renderer from "react-test-renderer";
import Text from "../";

it("renders correctly H1", () => {
  const tree = renderer.create(<Text.H1>test</Text.H1>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H1 darkGray", () => {
  const tree = renderer.create(<Text.H1 darkGray>test</Text.H1>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H1 lightGray", () => {
  const tree = renderer.create(<Text.H1 lightGray>test</Text.H1>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H2", () => {
  const tree = renderer.create(<Text.H2>test</Text.H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H2 darkGray", () => {
  const tree = renderer.create(<Text.H2 darkGray>test</Text.H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H2 lightGray", () => {
  const tree = renderer.create(<Text.H2 lightGray>test</Text.H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary", () => {
  const tree = renderer.create(<Text.Primary>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary darkGray", () => {
  const tree = renderer
    .create(<Text.Primary darkGray>test</Text.Primary>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary lightGray", () => {
  const tree = renderer
    .create(<Text.Primary lightGray>test</Text.Primary>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary", () => {
  const tree = renderer.create(<Text.Secondary>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary darkGray", () => {
  const tree = renderer
    .create(<Text.Secondary darkGray>test</Text.Secondary>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary lightGray", () => {
  const tree = renderer
    .create(<Text.Secondary lightGray>test</Text.Secondary>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary", () => {
  const tree = renderer.create(<Text.Tertiary>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary darkGray", () => {
  const tree = renderer
    .create(<Text.Tertiary darkGray>test</Text.Tertiary>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary lightGray", () => {
  const tree = renderer
    .create(<Text.Tertiary lightGray>test</Text.Tertiary>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
