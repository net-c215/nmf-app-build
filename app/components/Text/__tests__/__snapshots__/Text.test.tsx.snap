// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly H1 1`] = `
<Text.H1>
  test
</Text.H1>
`;

exports[`renders correctly H1 darkGray 1`] = `
<Text.H1
  darkGray={true}
>
  test
</Text.H1>
`;

exports[`renders correctly H1 lightGray 1`] = `
<Text.H1
  lightGray={true}
>
  test
</Text.H1>
`;

exports[`renders correctly H2 1`] = `
<Text.H2>
  test
</Text.H2>
`;

exports[`renders correctly H2 darkGray 1`] = `
<Text.H2
  darkGray={true}
>
  test
</Text.H2>
`;

exports[`renders correctly H2 lightGray 1`] = `
<Text.H2
  lightGray={true}
>
  test
</Text.H2>
`;

exports[`renders correctly H3 1`] = `
<Text.H3>
  test
</Text.H3>
`;

exports[`renders correctly H3 darkGray 1`] = `
<Text.H3
  darkGray={true}
>
  test
</Text.H3>
`;

exports[`renders correctly H3 lightGray 1`] = `
<Text.H3
  lightGray={true}
>
  test
</Text.H3>
`;

exports[`renders correctly Primary 1`] = `
<Text.Primary>
  test
</Text.Primary>
`;

exports[`renders correctly Primary darkGray 1`] = `
<Text.Primary
  darkGray={true}
>
  test
</Text.Primary>
`;

exports[`renders correctly Primary lightGray 1`] = `
<Text.Primary
  lightGray={true}
>
  test
</Text.Primary>
`;

exports[`renders correctly Secondary 1`] = `
<Text.Secondary>
  test
</Text.Secondary>
`;

exports[`renders correctly Secondary darkGray 1`] = `
<Text.Secondary
  darkGray={true}
>
  test
</Text.Secondary>
`;

exports[`renders correctly Secondary lightGray 1`] = `
<Text.Secondary
  lightGray={true}
>
  test
</Text.Secondary>
`;

exports[`renders correctly Tertiary 1`] = `
<Text.Tertiary>
  test
</Text.Tertiary>
`;

exports[`renders correctly Tertiary darkGray 1`] = `
<Text.Tertiary
  darkGray={true}
>
  test
</Text.Tertiary>
`;

exports[`renders correctly Tertiary lightGray 1`] = `
<Text.Tertiary
  lightGray={true}
>
  test
</Text.Tertiary>
`;
